app/javascript/react/containers/RestaurantFormContainer.js

```diff
@@ -0,0 +1,296 @@
+    this.submitRestaurant = this.submitRestaurant.bind(this);
+    this.handleFormSubmit = this.handleFormSubmit.bind(this);
+    this.formIsValid = this.formIsValid.bind(this);
+    this.onDrop = this.onDrop.bind(this);
+    this.successfulFormSubmit = this.successfulFormSubmit.bind(this);
+  }
+
+  successfulFormSubmit(event) {
+    event.preventDefault()
+    this.setState({
+        name: '',
+        street: '',
+        city: '',
+        state: '',
+        zip: '',
+        phoneNumber: '',
+        email: '',
+        website: '',
+        file: [],
+        errors: {},
+        notices: {}
+
+      })
+
+  }
+
+  formIsValid() {
+    let fields = ["name", "street", "city", "state", "zip", "phoneNumber"]
+    let errorFields = {}
+    let valid = true
+    fields.forEach((field) => {
+      if (this.state[`${field}`].trim() == '') {
+        let newError = {[field]: `You must enter a ${field}`}
+        errorFields = Object.assign(errorFields, newError)
+        valid = false
+      } else {
+        delete errorFields[`${field}`];
+      }
+
+    })
+    this.setState({ errors: errorFields })
+
+    return valid
+  }
+
+  handleFormSubmit(event) {
+    event.preventDefault();
+    console.log("handleFormSubmit")
+    if (this.formIsValid()) {
+        let body = new FormData()
+        body.append("name", this.state.name)
+        body.append("street", this.state.street)
+        body.append("city", this.state.city)
+        body.append("state", this.state.state)
+        body.append("zip", this.state.zip)
+        body.append("phoneNumber", this.state.phoneNumber)
+        body.append("email", this.state.email)
+        body.append("website", this.state.website)
+        body.append("photo", this.state.file[0])
+        body.append("cuisine_id", this.props.cuisine_id)
+        this.successfulFormSubmit(event)
+        this.submitRestaurant(body)
+      }
+
+  }
+
+  submitRestaurant(body) {
+    fetch(`/api/v1/restaurants`, {
+      method: 'POST',
+      credentials: 'same-origin',
+      body: body
+    }).then(response => {response
+      if (response.ok) {
+        return response
+      } else if (response.status == 422){
+        return response
+      } else {
+        let errorMessage = `${response.status} (${response.statusText})`,
+        error = new Error(errorMessage)
+        throw(error)
+      }
+    }).then(response => {
+      if (response.status == 422) {
+        console.log("code 422")
+        return response.json()
+      } else {
+
+        this.setState({notices: {notice: "Restaurant added successfully!"}})
+        return response.json()
+      }
+
+    }).then(parsedBody => {
+      if (parsedBody.errorList) {
+        this.setState({ errors: Object.assign(this.state.errors, parsedBody.errorList) })
+      } else {
+        this.props.updateParent(parsedBody)
+
+      }
+    })
+    .catch(error => console.error(`Error in fetch: ${error.message}`));
+  }
```

Comment:
Hmmm Looks like if the response is a 200 you will be calling setState twice here. A bit of a confusing web has been woving. take a crack at DRYing this up so that you dont call setState twice in a `.then`. If you cant I will allow this with warning.

Reply:
sure will do